action/action_abi.go

```diff
@@ -0,0 +1,38 @@
+package action
+
+import (
+	_ "embed" // import stakingContractABIJSON
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+var (
+	//go:embed staking_contract.abi
+	stakingContractABIJSON string
+	stakingContractABI     abi.ABI
+
+	candidateActivateMethod    abi.Method
+	candidateEndorsementMethod abi.Method
```

Comment:
break this file into `candidate_activate.go` and `candidate_endorsement.go`, just like other staking actions?
or move the ABI part in other staking actions into this file

Reply:
I think it's better to put all staking-related ABIs into one file, as this will detect conflicts between different parts of the ABI.

And added TODOs to move other parts into this file in other PRs